docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_instance_vpc.mdx

```diff
@@ -23,6 +23,265 @@ CISAWS, HIPAA, ISO27001
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Once you have AWS CLI installed and configured, you can start using it to interact with AWS services.
+
+2. To check Amazon SageMaker Notebook Instance in VPC, you can use the `describe-notebook-instance` command. This command returns information about a notebook instance. Here is an example of how to use this command:
+
+   ```
+   aws sagemaker describe-notebook-instance --notebook-instance-name my-notebook-instance
+   ```
+
+   Replace `my-notebook-instance` with the name of your notebook instance. This command returns a JSON object that contains information about the notebook instance.
+
+3. In the returned JSON object, look for the `DirectInternetAccess` field. If the value of this field is `Enabled`, it means that the notebook instance is not in a VPC. If the value is `Disabled`, it means that the notebook instance is in a VPC.
+
+4. To check if the notebook instance is in a DynamoDB, you can use the `describe-table` command. This command returns information about a table in DynamoDB. Here is an example of how to use this command:
+
+   ```
+   aws dynamodb describe-table --table-name my-table
+   ```
+
+   Replace `my-table` with the name of your table. This command returns a JSON object that contains information about the table. In the returned JSON object, look for the `Table` field. If the notebook instance is in the table, it should be listed in this field.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Amazon SageMaker Notebook Instances are in a VPC, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Setup AWS SDK for Python (Boto3):**
+   First, you need to set up Boto3. Install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   AWS_SESSION_TOKEN = 'your_session_token' # optional
+   ```
+
+2. **List all SageMaker Notebook Instances:**
+   Use the `list_notebook_instances` method to get all the notebook instances. Here is a sample script:
+   ```python
+   import boto3
+
+   client = boto3.client('sagemaker')
+
+   response = client.list_notebook_instances()
+
+   for instance in response['NotebookInstances']:
+       print(instance['NotebookInstanceName'])
+   ```
+
+3. **Check if the Notebook Instance is in a VPC:**
+   For each notebook instance, use the `describe_notebook_instance` method to get its details. If the `DirectInternetAccess` attribute is 'Disabled', then the notebook instance is in a VPC. Here is a sample script:
+   ```python
+   import boto3
+
+   client = boto3.client('sagemaker')
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>